src/datastore/postgres-store.ts

```diff
@@ -4053,6 +4056,44 @@ export class PgDataStore
     });
   }
 
+  async getTxStatus(txId: string): Promise<FoundOrNot<DbTxGlobalStatus>> {
+    return this.queryTx(async client => {
+      const chainResult = await client.query<DbTxGlobalStatus>(
+        `SELECT status, index_block_hash, microblock_hash
+        FROM txs
+        WHERE tx_id = $1 AND canonical = TRUE AND microblock_canonical = TRUE`,
```

Comment:
Is there anything to gain if we add a `LIMIT 1` here and in the the mempool query?